src/EditorFeatures/Core/Implementation/Formatting/FormatCommandHandler.TypeChar.cs

```diff
@@ -21,20 +21,15 @@ public VSCommanding.CommandState GetCommandState(TypeCharCommandArgs args, Func<
 
         public void ExecuteCommand(TypeCharCommandArgs args, Action nextHandler, CommandExecutionContext context)
         {
-            ExecuteCommand(args, nextHandler, CancellationToken.None);
+            _joinableTaskContext.Factory.Run(() => ExecuteReturnOrTypeCommand(args, nextHandler, CancellationToken.None));
         }
 
-        private void ExecuteCommand(TypeCharCommandArgs args, Action nextHandler, CancellationToken cancellationToken)
-        {
-            ExecuteReturnOrTypeCommand(args, nextHandler, cancellationToken);
-        }
-
-        private bool TryFormat(
+        private async Task<bool> TryFormatAsync(
             ITextView textView, Document document, IEditorFormattingService formattingService, char typedChar, int position, bool formatOnReturn, CancellationToken cancellationToken)
         {
-            var changes = formatOnReturn
-                ? formattingService.GetFormattingChangesOnReturnAsync(document, position, cancellationToken).WaitAndGetResult(cancellationToken)
-                : formattingService.GetFormattingChangesAsync(document, typedChar, position, cancellationToken).WaitAndGetResult(cancellationToken);
+            var changes = await (formatOnReturn
```

Comment:
❗️ The first line of this method needs to be the following:

```csharp
await _joinableTaskContext.Factory.SwitchToMainThreadAsync();
```

Reply:
reverted